From: nixnut <nixnut@gentoo.org>

KERNEXEC should probably only be enabled on x86 because otherwise
module.c will look for a header file that doesn't exist on most arches:

#ifdef CONFIG_PAX_KERNEXEC
#include <asm/desc.h>
#endif

Currently it is also enabled on ppc if the security level is set to
'high' (GRKERNSEC_HIGH).

KERNEXEC is supported on amd64/x86-64 in grsecurity upstream for
linux kernel 2.6.24.2.  This patch should therefore probably be
dropped for any hardened-sources-2.6.24 based upon grsecurity
patches for linux kernel 2.6.24.2.

Acked-by: Kerin Millar <kerframil@gmail.com>

--- a/grsecurity/Kconfig	2008-02-14 22:07:34.000000000 +0100
+++ b/grsecurity/Kconfig	2008-02-15 17:34:37.000000000 +0100
@@ -143,7 +143,7 @@
 	select PAX_EI_PAX
 	select PAX_PT_PAX_FLAGS
 	select PAX_HAVE_ACL_FLAGS
-	select PAX_KERNEXEC if (!X86_64 && !EFI && !COMPAT_VDSO && !PARAVIRT && X86_WP_WORKS_OK)
+	select PAX_KERNEXEC if (X86 && !X86_64 && !EFI && !COMPAT_VDSO && !PARAVIRT && X86_WP_WORKS_OK)
 	select PAX_MEMORY_UDEREF if (!X86_64 && !COMPAT_VDSO)
 	select PAX_RANDKSTACK if (X86_TSC && !X86_64)
 	select PAX_SEGMEXEC if (X86 && !X86_64)
